src/test/java/seedu/address/commons/util/StringUtilTest.java

```diff
@@ -77,6 +80,44 @@ public void isNonNegativeDoubleWithTwoOrLessDecimal() {
         assertTrue(StringUtil.isNonNegativeDoubleWithTwoOrLessDecimalPlaces("1"));
     }
 
+    //---------------- Tests for isNonNegativeInteger --------------------------------------
+
+    @Test
+    public void isNonNegativeInteger() {
+        // EP: empty strings
+        assertFalse(StringUtil.isNonNegativeInteger("")); // Boundary value
+        assertFalse(StringUtil.isNonNegativeInteger("  "));
+
+        // EP: not a number
+        assertFalse(StringUtil.isNonNegativeInteger("a"));
+        assertFalse(StringUtil.isNonNegativeInteger("aaa"));
+
+        // EP: zero
+        assertTrue(StringUtil.isNonNegativeInteger("0"));
+
+        // EP: negative zero
+        assertFalse(StringUtil.isNonNegativeInteger("-0"));
+
+        // EP: zero as prefix
+        assertTrue(StringUtil.isNonNegativeInteger("01"));
+
+        // EP: signed numbers
+        assertFalse(StringUtil.isNonNegativeInteger("-1"));
+        assertFalse(StringUtil.isNonNegativeInteger("+1"));
+
+        // EP: numbers with white space
+        assertFalse(StringUtil.isNonNegativeInteger(" 10 ")); // Leading/trailing spaces
+        assertFalse(StringUtil.isNonNegativeInteger("1 0")); // Spaces in the middle
+
+        // EP: number larger than Integer.MAX_VALUE
+        assertFalse(StringUtil.isNonNegativeInteger(Long.toString(Integer.MAX_VALUE + 1)));
+
+        // EP: valid numbers, should return true
+        assertTrue(StringUtil.isNonNegativeInteger("1")); // Boundary value
```

Comment:
Just a small nitpick, but I think 1 is not a boundary value? Since this is for non-negative integers, so the boundary values should be -1 and 0?